src/components/course/routes/ExternalCourseEnrollment.jsx

```diff
@@ -73,14 +69,17 @@ const ExternalCourseEnrollment = () => {
     }
   }, [externalCourseFormSubmissionError, containerRef]);
 
+  // Once a redemption has successfully completed and the can-redeem query has been invalidated or
+  // a user attempts to navigate directly to :slug/:courseType/course/:courseKey/enroll/:courseRunKey,
+  // redirect to the enrollment complete page.
   useEffect(() => {
-    // Once a redemption has successfully completed and the can-redeem query has been invalidated or
-    // a user attempts to navigate directly to :slug/:courseType/course/:courseKey/enroll,
-    //  it will run this conditional and perform the redirect
-    if (hasSuccessfulRedemption) {
+    const runHasSuccessfulRedemption = !!redeemabilityPerContentKey.find(
+      r => r.contentKey === courseRunKey,
+    )?.hasSuccessfulRedemption;
+    if (runHasSuccessfulRedemption) {
       navigate(completeEnrollmentUrl);
     }
-  }, [completeEnrollmentUrl, course.key, hasSuccessfulRedemption, navigate, pathname, slug]);
+  }, [completeEnrollmentUrl, course.key, courseRunKey, redeemabilityPerContentKey, navigate, pathname, slug]);
```

Comment:
I think `slug` is extraneous here?

Reply:
Yep, it does appear that way. Generally, only the variables used directly within the `useEffect` should be in the dependency array.